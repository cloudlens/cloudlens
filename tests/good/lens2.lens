lens messageLens(ident){
     stream (entry) when (entry) {
     	    print(ident + entry.message);
     }
}

{
	array = [{"message" : "bar 11"},
	      	 {"message" : "bar 22"},
		 {"message" : "bar 33"}];
}


{ messageLens("LENS " + ">>", array); }

stream (entry) {
       var ident = "STREAM >> ";
       print (ident + entry.message);
}

{ messageLens("LENS " + ">>");}

stream (entry) {
       var ident = "STREAM >> ";
       print (ident + entry.message);
}

