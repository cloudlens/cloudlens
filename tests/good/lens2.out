LENS >>bar 11
LENS >>bar 22
LENS >>bar 33
STREAM >> foo 1
STREAM >> foo 2
STREAM >> foo 3
STREAM >> foo 4
LENS >>foo 1
LENS >>foo 2
LENS >>foo 3
LENS >>foo 4
STREAM >> foo 1
STREAM >> foo 2
STREAM >> foo 3
STREAM >> foo 4
